crates/fluvio/src/producer/config.rs

```diff
@@ -88,7 +88,7 @@ pub struct TopicProducerConfig {
     pub(crate) linger: Duration,
     /// Partitioner assigns the partition to each record that needs to be send
     #[builder(default = "default_partitioner()")]
-    pub(crate) partitioner: Box<dyn Partitioner + Send + Sync>,
+    pub(crate) partitioner: Arc<dyn Partitioner + Send + Sync>,
```

Comment:
Perhaps we should make this Arc Box?  It may be important in long running or high concurrency uses

Reply:
I don't think so.
In our usecase that we will only clone once, in the builders, I think that we will not have a overhead here.

Maybe we can use [dyn_clone](https://docs.rs/dyn-clone/latest/dyn_clone/) and keep using box. That allows us to have clone and use box.
But using Arc here does not seem a problem.

Reply:
Given that arc uses heap memory from the internals that you researches, the current Arc return is fine!